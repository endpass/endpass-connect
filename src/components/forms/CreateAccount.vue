<template>
  <div>
    <message>
      You have not any account. Create one and you will be logged in.
    </message>
    <form-controls>
      <v-button type="primary" data-test="submit-button" @click="emitRequest">
        Create account
      </v-button>
    </form-controls>
  </div>
</template>

<script>
import VFrame from '../VFrame.vue';
import VInput from '../VInput.vue';
import VButton from '../VButton.vue';
import Message from '../Message.vue';
import FormControls from '../FormControls.vue';

export default {
  name: 'CreateAccountForm',

  data: () => ({
    email: '',
  }),

  methods: {
    emitRequest() {
      this.$emit('request');
    },
  },

  components: {
    VFrame,
    VButton,
    VInput,
    Message,
    FormControls,
  },
};
</script>
