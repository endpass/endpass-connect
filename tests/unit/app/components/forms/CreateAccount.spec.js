import { shallowMount, mount } from '@vue/test-utils';
import CreateAccount from '@/components/forms/CreateAccount.vue';

describe('CreateAccount', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(CreateAccount, {
        propsData: {},
      });
    });

    it('should correctly render CreateAccount component', () => {
      expect(wrapper.name()).toBe('CreateAccountForm');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(CreateAccount, {
        propsData: {},
      });
    });

    describe('submit feature', () => {
      it('should emit request on click submit button by default', () => {
        wrapper.find('[data-test=submit-button]').trigger('click');

        expect(wrapper.emitted().request).toBeTruthy();
      });
    });
  });
});
