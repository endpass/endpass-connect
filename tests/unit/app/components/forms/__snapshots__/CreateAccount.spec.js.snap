// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount render should correctly render CreateAccount component 1`] = `
<div>
  <message-stub>
    You have not any account. Create one and you will be logged in.
  </message-stub>
  <form-controls-stub>
    <v-button-stub type="primary" data-test="submit-button">
      Create account
    </v-button-stub>
  </form-controls-stub>
</div>
`;
